docs/proposals/20210526-cluster-class-and-managed-topologies.md

```diff
@@ -194,588 +185,116 @@ If instead you are eager to see an example of ClusterClass and how the Cluster o
-  - `spec.topology.version` cannot be unset and must be a valid semver, if being updated.
-  - `spec.topology.version` cannot be downgraded.
-  - `spec.topology.workers.machineDeployments[i].name` field must be unique within a Cluster
-  - A set of worker nodes can be added to or removed from the `spec.topology.workers.machineDeployments` list.
-  - (defaulting) variables are defaulted according to the corresponding `ClusterClassVariable`
-  - all required variables must exist and match the schema defined in the corresponding `ClusterClassVariable` in the ClusterClass
-  - (defaulting) nested fields of `spec.topology.workers.machineDeployments[i].variables.overrides` are defaulted according to the corresponding `ClusterClassVariable`
-  - `spec.topology.workers.machineDeployments[i].variables.overrides` must match the schema defined in the corresponding `ClusterClassVariable` in the ClusterClass
-  
-#### ClusterClass compatibility
+#### Validation and Defailting
+
+Both the new field in the Cluster CRD and the new ClusterClass CRD type will be validated according to what is specified in the
+API definitions; additionally please consider the following:
 
-There are cases where we must consider whether two ClusterClasses are compatible:
-1. Where a user chooses to replace an existing ClusterClass `cluster.spec.topology.class` with a new ClusterClass.
-2. Where a user updates a ClusterClass currently in use by a Cluster.
+**ClusterClass**
 
-To establish compatibility between two ClusterClasses:
-  - All the references must be in the same namespace of `metadata.Namespace` - the same namespace as the existing clusterClass.
-  - `spec.workers.machineDeployments` must not remove any deployment classes (adding new or modifying existing classes is supported).
-  - `spec.controlPlane.localobjecttemplate`, `spec.controlplane.machineinfrastructure`, `spec.infrastructure`,  `spec.workers.machineDeployments[].template.infrastructure.ref` must not change apiGroup or Kind.
-  - All variables used in Clusters must not be removed.
-  - Schemas of variables used in Clusters must not be changed in an incompatible way, i.e. when the schema change would be reconciled, 
-    applying patches to templates should not lead to incompatible changes when reconciling the objects inferred from those templates.
+- It is not allowed to change apiGroup or Kind for the referenced templates (with the only exception of the bootstrap templates).
+- MachineDeploymentClass cannot be removed as long as they are used in Clusters.
+- It’s the responsibility of the ClusterClass author to ensure the patches are semantically valid, in the sense they
+  generate valid templates for all the combinations of the corresponding variables in input.
+- Variables cannot be removed as long as they are used in Clusters.
+- When changing variables definitions, the system validate schema changes against existing clusters and blocks in case the changes are 
```

Comment:
```suggestion
- When changing variable definitions, the system validates schema changes against existing clusters and blocks in case the changes are 
```